src/cpp/include/openvino/genai/perf_metrics.hpp

```diff
@@ -49,6 +54,16 @@ struct OPENVINO_GENAI_EXPORTS MeanStdPair {
     float std;
 };
 
+/**
+* @brief Structure to store list of durations in milliseconds.
+*/
+struct OPENVINO_GENAI_EXPORTS SummaryStats {
+    float mean;
```

Comment:
Why not `MicroSeconds`?
Does it make sense to summaries compile time? The time highly depends on what's compiled

Reply:
We use float as well for `MeanStdPair`, so that final user don't have to cast manually. 

Here is `SummaryStats` instead of `MeanStdPair`  because need more fine grained values.

It usually it differs highly on (yes you are right) on what is compiled and whether it's compiled for the first time. Example below shows times for the exactly the same grammar (around 1000 json fields), at first it's slow but subsequent ones after much faster because of the caching.

Other values such as tpot, thoughtput, etc. do not change that drastically. After discussion with @apaniukov we decided that it's very handy to see that hight and lows right away, without taking look into raw metrics.

![image](https://github.com/user-attachments/assets/0ecd6e7c-bcfd-455e-bda5-56278f1d6e4e)

Reply:
For the users of xgrammar it's all the same compile call. `m_grammar_compiler->CompileGrammar(grammar)`, caching happens inside the grammar object `std::unique_ptr<xgrammar::GrammarCompiler> m_grammar_compiler`.

In the example above `[5056.2, 10.7, 10.4, 10.5, 10.3]` (converted from microsec to ms for simplicity) if i get only mean and std It would be mean `1000 ms`, std `~2000 ms`. If we wanna know for example how much time it took to compile for the first time this numbers doesn't give us fine grained info.

But when we have min and max we can get the first initial compilation time which is max `5 000 ms`, and subsequent compilation times min `10 ms`.